### ISSUE TITLE
Issues testing react app with stencil built library

### ISSUE THREAD

**Issue by corysmc** [nxext/nx-extensions#235]
**Describe the bug**
Simple test in react app that imports a stencil generated component fails because of a syntax error when importing the loader file.

**To Reproduce**
1. clone this repo https://github.com/corysmc/nx-stencil-workspace/tree/31842c0d3d0cbfc3964b7a04dd53677ce6dcf20f (basic setup for stencil components, react application that consumes those components with the loader included): 
2. `npm i`
3. `nx build core`
4. `nx test ionic-react-app`
5. Test fails trying to import the loader file in core-react

```Test suite failed to run

    SyntaxError: <...>/nx-stencil-workspace/dist/libs/core/loader/index.d.ts: Support for the experimental syntax 'flow' isn't currently enabled (3:8):

      1 |
      2 | export * from '../dist/types/components';
    > 3 | export interface CustomElementsDefineOptions {
        |        ^
      4 |   exclude?: string[];
      5 |   resourcesUrl?: string;
      6 |   syncQueue?: boolean;
```

---

**Workaround 1 (kind of)**
1. Find the core-react/src/index.ts file (generated after running nx build core)
2. Change the import from:
    `import { defineCustomElements } from '@my-org/core/loader';` 
    to 
    `import { defineCustomElements } from '../../../dist/libs/core/loader';`
3. run `nx test ionic-react-app --skip-nx-cache`
4. Test paseses.

So something is going wrong here with jest. It's importing the typescript file definition, instead of the index.js file from the loader. I've searched high and low and tried several changes to tsconfig, babelrc, jest.presets, etc... without success.

---

**Workaround 2**
The other thing that seems to work, is 
1. copying the dist folder of `dist/libs/core` and 
2. pasting into `node_modules/@my-org/core` to "trick" jest into realizing this is a module and it should use the correct js file.

---

**Workaound 3**
1. Change your `stencil.config.ts` file to not include the define custom elements (loader) script
```reactOutputTarget({
      componentCorePackage: '@my-org/core',
      proxiesFile: '../../../libs/core-react/src/index.ts',
      includeDefineCustomElements: false,
    }),
```
2. Include the scripts [outlined on stenciljs](https://stenciljs.com/docs/react) to manually include the loader in your react app in `main.tsx`
```
import { applyPolyfills, defineCustomElements } from '@my-org/core/loader';

ReactDOM.render(<App />, document.getElementById('root'));

applyPolyfills().then(() => {
  defineCustomElements();
});
```

**Expected behavior**
Have the correct settings when initializing a project with `@nxext/stencil` so that this doesn't happen
Ideally I could have the stencil library generate the loader file, and just import the components into the projects I want without jest issues when running tests.

**Comment by bigfish** [comment]
Yeah, I've been battling with this as well for some time now :( Finally find the minimal repro steps as outlined. 
Thanks for posting the workarounds @corysmc I'll probably have to go with one of them until this is resolved, seems like the plugin may need an update.
